### ISSUE TITLE
Wrong closure capture in folded cycle

### ISSUE THREAD

**Issue by VoronFX** [bridgedotnet/Bridge#3131]
### Steps To Reproduce

https://deck.net/a2d9df1ab9da9e244785d985e6187fe3

```c#
public class Program
{
    public static int[] data = new int[]{ 1, 2, 3};
    public static List<Action> actions = new List<Action>();
    
    public static void Main()
    {
       	foreach	(var i2 in data)
        foreach	(var i in data)
            actions.Add(()=> Console.WriteLine(i));
        
        foreach (var action in actions)
            action();
    }
}
```

### Expected Result

```js
1
2
3
1
2
3
1
2
3
```

### Actual Result

```js
3
3
3
3
3
3
3
3
3
```

**Comment by geoffreymcgill** [comment]
Hi @VoronFX – Thanks for reporting this issue, and for the excellent sample demonstrating how to reproduce. We will investigate right away and fix.

**Comment by VoronFX** [comment]
No problem! I've discovered it while preparing a speech about Bridge.NET awesomeness for upcoming .NET Developers Meetup in Tokyo. =)

**Comment by geoffreymcgill** [comment]
Cool! You should post an announcement for the Tokyo meetup in the [fourms](https://forums.bridge.net/forum/general/announcements).

I would like to know more, and please feel free to contact me directly (geoff@object.net) if you require any assistance.

**Comment by LeonidVeriga** [comment]
This issue has been fixed and will be included in the upcoming Bridge 16.3.1 release.
